compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -641,6 +641,39 @@ pub fn super_relate_consts<'tcx, R: TypeRelation<'tcx>>(
                 ty: a.ty(),
             }));
         }
+        (ty::ConstKind::Abstract(ae), ty::ConstKind::Abstract(be)) => {
+            let r = relation;
+            // FIXME(julianknodt): is it possible to relate two consts which are not identical
+            // exprs? Should we care about that?
+            let expr = match (ae, be) {
+                (Expr::Binop(a_op, al, ar), Expr::Binop(b_op, bl, br)) if a_op == b_op => {
+                    Expr::Binop(
+                        a_op,
+                        super_relate_consts(r, al, bl)?,
```

Comment:
`r.consts(al, bl)?` you want to give the `TypeRelation` a chance to interact with the nested constant.